snooty/parser.py

```diff
@@ -616,7 +618,19 @@ def handle_directive(
                     self.validate_list_table(bullets, expected_num_columns)
 
         elif name == "openapi":
+            # Parsing should be done by the OpenAPI renderer on the frontend by default
+            snooty_parse = options.get("snooty-parse", False)
+
             if argument_text is None:
+                # Check if argument is a url instead
+                url_argument = None
+                try:
+                    url_argument = argument[0].children[0].value
```

Comment:
Interesting, I'm just wondering — how do we know this is not a filepath?

Reply:
Good point! I can change this logic to check `argument[0].refuri`, since `argument[0]` should be a `reference` if the argument is a url.

I've only ever tested this using a filepath or a url as an argument. A filepath would be assigned as `argument_text` and a url would typically be a reference with a text node with the same url as its value.

Reply:
Ah, thanks for the explanation. I think checking for the `refuri` probably makes sense here, good idea!